notebooks/sequence_and_schema.ipynb

```diff
@@ -0,0 +1,172 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Data sequence and schema diagrams\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Data Sequences\n",
+    "### File apis\n",
+    "\n",
+    "```mermaid\n",
+    "---\n",
+    "title: Data sequence - POST /upload\n",
+    "---\n",
+    "\n",
+    "sequenceDiagram\n",
+    "    Django->>S3: file key, content\n",
+    "    Django->>Core: file key\n",
+    "    Core->>Workers: file key\n",
+    "    Core->>Elastic: file key\n",
+    "    S3->>Workers: file content\n",
+    "    Workers->>Elastic: chunk key, content\n",
+    "```\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Chat APIs\n",
+    "\n",
+    "\n",
+    "```mermaid\n",
+    "---\n",
+    "title: Data sequence - POST /chat/vanilla\n",
+    "---\n",
+    "\n",
+    "sequenceDiagram\n",
+    "    Django->>Core: ChatHistory.messages[]\n",
+    "    Core->>LLM API: ChatHistory.messages[]\n",
+    "\n",
+    "```\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```mermaid\n",
+    "---\n",
+    "title: Data sequence - POST /chat/rag\n",
+    "---\n",
+    "\n",
+    "sequenceDiagram\n",
+    "    Django->> Core: ChatHistory.messages[], File[].uuid\n",
+    "    Core->>Embedding API: ChatHistory.messages[]\n",
+    "    Embedding API->>Core: ChatHistory.messages[].embeddings\n",
+    "    Elastic->>Core: File[].Chunk[].embeddings\n",
+    "    Core->>LLM API: ChatHistory.messages[].embeddings + File[].Chunk[].embeddings\n",
+    "\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Schema\n",
+    "\n",
+    "```mermaid\n",
+    "---\n",
+    "title: Django schema\n",
+    "---\n",
+    "\n",
+    "erDiagram\n",
+    "    User }|--|{ UserGroup : \"UserGroup.users\"\n",
+    "    User {\n",
+    "        UUID uuid\n",
+    "        string name\n",
+    "    }\n",
+    "    UserGroup {\n",
+    "        UUID uuid\n",
+    "        string name\n",
+    "        UUID[] users\n",
+    "    }\n",
+    "    \n",
+    "    FileRecord }|--|| UserGroup: \"FileRecord.owner\"\n",
+    "    FileRecord {\n",
+    "        UUID uuid\n",
+    "        UUID owner\n",
+    "        string key \n",
+    "    }\n",
+    "\n",
+    "    UserGroup ||--|{ Message : \"Mesage.owner\"\n",
+    "    Message {\n",
+    "        UUID uuid\n",
+    "        UUID owner\n",
+    "        string text\n",
+    "        string role \n",
+    "    }\n",
+    "\n",
+    "    Message }|--|| ChatHistory: \"ChatHistory.messages\"\n",
+    "    UserGroup ||--|{ ChatHistory: \"ChatHistory.owner\"\n",
+    "\n",
+    "    ChatHistory {\n",
+    "        UUID uuid\n",
+    "        string name\n",
+    "        UUID owner\n",
+    "        UUID[] messages\n",
```

Comment:
given that these relationships are 1:* should we move the reference to the child class, i.e. Message & File have references to ChatHistory?

Reply:
lets rename `files_received` to `files_retrieved` (to match up with RAG wording).

George, could you explain that referencing? Not sure I grasp the concept

Reply:
> ...explain that referencing...

its better/easier to store 1 parent-reference than many child-references?

Reply:
Ah, yes one parent reference. That'll match up with our File Chunk style relationship

Reply:
Yes to messages, not to files (which could be used in several chats) - but that's my mistake of labelling